_studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy_defaults.cpp

```diff
@@ -1438,15 +1438,23 @@ class GetDefault
         , const mfxVideoParam& par
         , eMFXHWType hw)
     {
+        auto fcc = par.mfx.FrameInfo.FourCC;
+
         bool bValid =
             par.mfx.LowPower
             && !CheckTriState(par.mfx.LowPower);
 
         bool bOn =
-            (hw == MFX_HW_CNL
-                && par.mfx.TargetUsage >= 6
-                && par.mfx.GopRefDist < 2
-                && !bValid);
+            (!bValid &&
+             ((hw == MFX_HW_CNL
+               && par.mfx.TargetUsage >= 6
+               && par.mfx.GopRefDist < 2) ||
+              (hw >= MFX_HW_ICL &&
+               (fcc == MFX_FOURCC_AYUV
+#if (MFX_VERSION >= 1027)
```

Comment:
Looks like #if is not needed, MFX_FOURCC_Y410 is already used without it.

Reply:
Disagree here, this is how it defined in header:
```
#if (MFX_VERSION >= 1027)
    MFX_FOURCC_Y210         = MFX_MAKEFOURCC('Y','2','1','0'),
    MFX_FOURCC_Y410         = MFX_MAKEFOURCC('Y','4','1','0'),
#endif
```

If you know some places where Y410 used unprotected, please open issue ticket.

Reply:
I insist. There are huge amount of use. MFX_VERSION is used to to control enabling new features, but not  every other name definition. It would make code unreadable. It is assumed to be true. At least removing it will not break anything and match other code.

Reply:
@lakulako what do you mean by other name definition? ```MFX_VERSION``` used to give ability to compile with certain API version. I should be able to compile latest code with API 1.25 for example, and all unsupported in API 1.25 features should be turned off. If you pull ```MFX_FOURCC_Y410``` out of define here, code will not compile.

Reply:
@mgonchar, I meant every 2nd name in the code. There are already many names used without such #if. Including MFX_FOURCC_Y410 in the same module. The code will not compile with such old MFX versions neither with nor without this #if. Compilation with old mfx versions is not an option. In the development #if (MFX_VERSION >= MFX_VERSION_NEXT) is used. After VERSION_NEXT is advanced, obsolete #if are deleted from code.
If one needs older mfx, he can checkout older version from github.

Reply:
>> After VERSION_NEXT is advanced, obsolete #if are deleted from code.

No, it is done in other way, those ```#if``` remains in code with correct version of API. And all places which violates such rule should be fixed.

@onabiull I believe we need your input here.

Reply:
The only user known to me which gets use of older API versions built from the same source code base is Android:
https://github.com/Intel-Media-SDK/MediaSDK/blob/master/android/mfx_defs.mk#L54

Just looking into their build scripts, we get:
* Android O: 1.26
* Android P: 1.30
* everything else: 1.32

No idea which build paths are still alive from support perspective though...

I can't say that I very much like approach we use to support API versions and library code corresponding to them, but that's what we are doing as of toda. And to start doing otherwise at this point requires considerations.

Reply:
@mgonchar @dvrogozh @lakulako Thanks for your comments, so here Y410 should be used with protection, right?

Reply:
> @mgonchar @dvrogozh @lakulako Thanks for your comments, so here Y410 should be used with protection, right?

2 cents: 
IMHO the fact that we're putting feature definitions under `#if SOME_VERSION` is an extra burden (especially from implementation point of view: we have to put implementations under corresponding `#if SOME_VERSION` ) without real benefits. I don't know real cases where it's really (objectively!) required.  
BUT we _already_ have it, so in order to be consistent, we have either to support it or fully drop. 
So my answer is **yes**, we need the protection there.

Reply:
@dmitryermilov thanks for confirming this.